### PR TITLE
[datadog_dashboard_list] Migrate to tf framework

### PR DESCRIPTION
**Pull request by HantingZhang2**

### CODE REVIEW

File: datadog/fwprovider/resource_datadog_dashboard_list.go [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+type dashboardListResourceModel struct {
+	ID       types.String     `tfsdk:"id"`
+	Name     types.String     `tfsdk:"name"`
+	DashItem []*DashItemModel `tfsdk:"dash_item"`
+}
+
+type DashItemModel struct {
+	Type    types.String `tfsdk:"type"`
+	Dash_id types.String `tfsdk:"dash_id"`
```
Comment (skarimo) [inline datadog/fwprovider/resource_datadog_dashboard_list.go:41]:
```suggestion
	DashID types.String `tfsdk:"dash_id"`
```

File: datadog/fwprovider/resource_datadog_dashboard_list.go [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+	Auth  context.Context
+}
+
+type dashboardListResourceModel struct {
+	ID       types.String     `tfsdk:"id"`
+	Name     types.String     `tfsdk:"name"`
+	DashItem []*DashItemModel `tfsdk:"dash_item"`
+}
+
+type DashItemModel struct {
```
Comment (skarimo) [inline datadog/fwprovider/resource_datadog_dashboard_list.go:39]:
```suggestion
type dashItemModel struct {
```

**Review by skarimo — COMMENTED** [review]
This looks great overall. However, we need to consider the issue https://github.com/DataDog/terraform-provider-datadog/issues/1979 . Previously, the sdk did not care about mismatch between state and config after a terraform action was performed but with this change, it will exit with `Provider produced inconsistent result after apply`.

I think the best approach here is to modify the updateState methods to always use whatever the user supplied. Something like:

```
func (r *dashboardListResource) updateStateFromResponse(ctx context.Context, state *dashboardListResourceModel, dashboards []datadogV2.DashboardListItemResponse) {
	temp := []*DashItemModel{}
	for _, dashboard := range dashboards {
		dashboardItem := DashItemModel{}
		dashboardItem.Dash_id = types.StringValue(dashboard.GetId())

		found := false
		for _, v := range state.DashItem {
			if v.Dash_id.ValueString() == dashboard.GetId() {
				dashboardItem.Type = v.Type
				found = true
			}
		}
		if !found {
			dashboardItem.Type = types.StringValue(string(dashboard.GetType()))
		}

		temp = append(temp, &dashboardItem)
	}
	state.DashItem = temp
}
```

File: datadog/fwprovider/resource_datadog_dashboard_list.go [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+			},
+			"id": utils.ResourceIDAttribute(),
+		},
+		Blocks: map[string]schema.Block{
+			"dash_item": schema.SetNestedBlock{
+				Description: "A set of dashboard items that belong to this list",
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"type": schema.StringAttribute{
+							Description: "The type of this dashboard. Valid values are `custom_timeboard`, `custom_screenboard`, `integration_screenboard`, `integration_timeboard`, `host_timeboard`.",
```
Comment (skarimo) [inline datadog/fwprovider/resource_datadog_dashboard_list.go:75]:
```suggestion
							Description: "The type of this dashboard.
```

**Review by skarimo — COMMENTED** [review]
Change to be applied after https://github.com/DataDog/terraform-provider-datadog/pull/2042

**Review by nkzou — APPROVED** [review]

**Review by skarimo — APPROVED** [review]

**Pull request merged**